packages/generator/ni_measurement_plugin_sdk_generator/client/_support.py

```diff
@@ -344,9 +345,19 @@ def extract_base_service_class(service_class: str) -> str:
     return base_service_class
 
 
-def create_module_name(base_service_class: str) -> str:
+def create_module_name(base_service_class: str, directory: pathlib.Path) -> str:
     """Creates a module name using base service class."""
-    return _camel_to_snake_case(base_service_class) + "_client"
+    base_module_name = _camel_to_snake_case(base_service_class) + "_client"
+    module_name_with_extension = f"{base_module_name}.py"
+    filepath = os.path.join(directory, module_name_with_extension)
+    counter = 1
+
+    while os.path.exists(filepath):
+        module_name_with_extension = f"{base_module_name}_{counter}.py"
```

Comment:
Adding an underscore before the counter looks a little weird.

Reply:
I have removed the underscore.